cmake/macos/compilerconfig.cmake

```diff
@@ -15,6 +15,18 @@ if(NOT CMAKE_OSX_ARCHITECTURES)
 endif()
 set_property(CACHE CMAKE_OSX_ARCHITECTURES PROPERTY STRINGS arm64 x86_64)
 
+message(DEBUG "macOS SDK Path: ${CMAKE_OSX_SYSROOT}")
+string(REGEX MATCH ".+/MacOSX.platform/Developer/SDKs/MacOSX([0-9]+\.[0-9])+\.sdk$" _ ${CMAKE_OSX_SYSROOT})
+set(OBS_MACOS_SDK_CURRENT ${CMAKE_MATCH_1})
+message(DEBUG "macOS SDK version: ${OBS_MACOS_SDK_CURRENT}")
+set(OBS_MACOS_SDK_MINIMUM "13.1")
+if(OBS_MACOS_SDK_CURRENT VERSION_LESS OBS_MACOS_SDK_MINIMUM)
+  message(
+    FATAL_ERROR
+      "Your macOS SDK version (${OBS_MACOS_SDK_CURRENT}) is too low. The macOS ${OBS_MACOS_SDK_MINIMUM} SDK is required to build OBS."
```

Comment:
Maybe mention the minimum Xcode version that is required to be installed to fix this issue.

Reply:
Would that require us to manually update this message each time or separately track the minimum Xcode version?

Reply:
Both Xcode and macOS SDK Version would need to be updated here.
We could add this to buildspec or the default cmake preset but I’m not sure it makes sense in either? cc @PatTheMav

Reply:
Preset would be best suited, it's where we set the deployment target as well.

Reply:
Added as a preset variable.

Reply:
Reverted this change.